### PR TITLE
docs: adds callouts linking to Gatsby Cloud

### PR DESCRIPTION
**Pull request by gillkyle**
## Description

This pull request adds a `CloudCallout` component and links it in docs that serve as logical stepping stones into things Gatsby Cloud helps solve.

image

There is also an additional call to action added on the homepage that keeps the primary action linking users to the docs. For those already using Gatsby and looking for Cloud this creates a more bi-directional flow between Gatsby Cloud and the docs.

image

## Related Issues

_None_

### CODE REVIEW

File: docs/contributing/docs-and-blog-components.md [hunk 1]
```diff
@@ -124,6 +124,38 @@ Rendered, the component looks like this:
   To improve is to change, so to be perfect is to have changed often.
 </Pullquote>
 
+### Gatsby Cloud Callout
+
+The `<CloudCallout />` component is used to call attention and link to Gatsby Cloud. It wraps a small call to action in styles that make content stand out.
+
+#### Usage
+
+The Cloud Callout component takes one optional props, and prepends the children it wraps to the general call to action:
```
Comment (laurieontech) [inline docs/contributing/docs-and-blog-components.md:133]:
```suggestion
The Cloud Callout component takes one optional prop, and prepends the children it wraps to the general call to action:
```

**Comment by gillkyle** [comment]
Thanks @laurieontech! I didn't check dark mode, those were great points. I made the text match the size of the rest of the page, used a purple that has more contrast, updated that svg circle to remove the fill since it really stands out in dark mode. 

I also fixed that mistake with the plural props 🙂

**Comment by laurieontech** [comment]
Looks great! The only slightly weird thing is that the `Gatsby Cloud` link is no longer emphasized in light mode. Intentional?
Screen Shot 2020-04-21 at 4 35 10 PM

**Comment by gillkyle** [comment]
That's a very valid point, I actually don't see any huge reason to make all that other text purple. 

I'll updated it so just the link text shows as purple.

image

image

**Comment by laurieontech** [comment]
Looks good! Sorry for being so nitpicky.

**Pull request merged**